### PR TITLE
NEXT-10855 - Add ability to search by Custom Fields in Admin Backend

### PR DESCRIPTION
**Pull request by MrocznyCzesiek**
### 1. Why is this change necessary?

It gives possibility to search by Custom Fields in Administration.

### 2. What does this change do, exactly?

It adds `SearchRanking` flag which is used by `EntityScoreQueryBuilder` to build queries based on the Term provided.

### 3. Describe each step to reproduce the issue or behaviour.

Example: Please create i.e. a product with Custom Field. Go to search field in Administration, fill in the Custom Field value (or part) and hit ENTER. The product should be returned in search results. It works for Products, Manufacturers and Customers in Administration only.

### 5. Checklist

- [x] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [x] I have created a [changelog file](https://github.com/shopware/platform/blob/master/adr/2020-08-03-Implement-New-Changelog.md) with all necessary information about my changes
- [x] I have written or adjusted the documentation according to my changes
- [x] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: changelog/_unreleased/2020-09-17-add-ability-to-search-by-custom-field-in-administration-backend.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+title: Add ability to search by Custom Fields in Administration Backend
+issue: NEXT-10855
+author: Mroczny Czesiek, Wojciech Milek
+author_email: mrocznyczesiek@gmail.com, wojciechus@hotmail.com
```
Comment (PaddyS) [inline changelog/_unreleased/2020-09-17-add-ability-to-search-by-custom-field-in-administration-backend.md:5]: Please choose one of them :-)

File: changelog/_unreleased/2020-09-17-add-ability-to-search-by-custom-field-in-administration-backend.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+title: Add ability to search by Custom Fields in Administration Backend
+issue: NEXT-10855
+author: Mroczny Czesiek, Wojciech Milek
+author_email: mrocznyczesiek@gmail.com, wojciechus@hotmail.com 
+author_github: @MroczyCzesiek, @wojciechus
+---
+# Core
+* Added `SearchRanking` flag to `customFields` in `ProductDefinition`, `CustomerDefinition` and `ProductManufacturerDefinition`
+* It has no impact on the Storefront search functionality
+* Tested on the shop with 6000+ products - no significant impact on search performance
```
Comment (PaddyS) [inline changelog/_unreleased/2020-09-17-add-ability-to-search-by-custom-field-in-administration-backend.md:11]: Only keywords `Added`, `Changed`, `Removed` and `Deprecated` are allowed here

File: changelog/_unreleased/2020-09-17-add-ability-to-search-by-custom-field-in-administration-backend.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+---
+title: Add ability to search by Custom Fields in Administration Backend
+issue: NEXT-10855
+author: Mroczny Czesiek, Wojciech Milek
```
Comment (PaddyS) [inline changelog/_unreleased/2020-09-17-add-ability-to-search-by-custom-field-in-administration-backend.md:4]: Unfortunately we only support one author at the moment, also for `author_email` and `author_github`

File: src/Core/Content/Product/ProductDefinition.php [hunk 4]
```diff
@@ -173,7 +173,7 @@ protected function defineFields(): FieldCollection
             (new TranslatedField('metaTitle'))->addFlags(new Inherited()),
             (new TranslatedField('packUnit'))->addFlags(new Inherited()),
             (new TranslatedField('packUnitPlural'))->addFlags(new Inherited()),
-            (new TranslatedField('customFields'))->addFlags(new Inherited()),
+            (new TranslatedField('customFields'))->addFlags(new Inherited(), new SearchRanking(SearchRanking::HIGH_SEARCH_RANKING)),
```
Comment (OliverSkroblin) [inline src/Core/Content/Product/ProductDefinition.php:176]: i would set a MIDDLE_SEARCH_RANKING

Comment (wojciechus) [reply src/Core/Content/Product/ProductDefinition.php:176]: only here or in each three places?

Comment (OliverSkroblin) [reply src/Core/Content/Product/ProductDefinition.php:176]: in all places ;)

Comment (MrocznyCzesiek) [reply src/Core/Content/Product/ProductDefinition.php]: I've just fixed the ranking

File: changelog/_unreleased/2020-09-17-add-ability-to-search-by-custom-field-in-administration-backend.md (hunk 3 above)
Comment (PaddyS) [reply changelog/_unreleased/2020-09-17-add-ability-to-search-by-custom-field-in-administration-backend.md:4]: This still needs to be done for the `author`, which this comment was originally pointing at. Sorry for any confusion!

**Review by shyim — APPROVED** [review]

**Pull request merged**